### PR TITLE
Support for storing multiple InResponseTo IDs in the http session

### PR DESCRIPTION
**Pull request by DinikaSen**
### Proposed changes in this pull request

This PR introduces the following changes.
- Fix ID validation failures that occur intermittently, and when the browser is refreshed after login into the sample app.
- Fix the ID validation failure error message to give the actual reason for the failure.

### CODE REVIEW

File: io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java [hunk 1]
```diff
@@ -377,23 +377,37 @@ private String createBaseFormPage(String params, String redirectURI) {
...
 
     private void setIDtoSession(HttpSession httpSession, String saml2RequestId) {
 
-        httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_NAME, saml2RequestId);
+        if (httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME) != null) {
+            ArrayList<String> samlRequestIDs =
+                    (ArrayList<String>) httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME);
+            if (!samlRequestIDs.contains(saml2RequestId)) {
+                samlRequestIDs.add(saml2RequestId);
+                httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME, samlRequestIDs);
```
Comment (ChamathNS) [inline io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java:404]: Can't we have this outside the if else block?

File: io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java [hunk 2]
```diff
@@ -377,23 +377,37 @@ private String createBaseFormPage(String params, String redirectURI) {
...
         }
     }
 
     private void setIDtoSession(HttpSession httpSession, String saml2RequestId) {
 
-        httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_NAME, saml2RequestId);
+        if (httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME) != null) {
+            ArrayList<String> samlRequestIDs =
+                    (ArrayList<String>) httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME);
+            if (!samlRequestIDs.contains(saml2RequestId)) {
```
Comment (ChamathNS) [inline io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java:402]: In the event `httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME)` returns null, will `samlRequestIDs.contains(saml2RequestId)` be null safe?

Comment (DinikaSen) [reply io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java:402]: This code segment is written within a `if (httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME) != null)` segment.

File: io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java [hunk 3]
```diff
@@ -377,23 +377,37 @@ private String createBaseFormPage(String params, String redirectURI) {
...
+                samlRequestIDs.add(saml2RequestId);
+                httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME, samlRequestIDs);
+            }
+        } else {
+            ArrayList<String> samlRequestIDs = new ArrayList<>();
+            samlRequestIDs.add(saml2RequestId);
+            httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME, samlRequestIDs);
+        }
```
Comment (ChamathNS) [inline io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java:410]:
```suggestion
        ArrayList<String> samlRequestIDs = new ArrayList<>();
        if (httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME) != null) {
            samlRequestIDs = (ArrayList<String>) httpSession.getAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME);
        } 
        if (!samlRequestIDs.contains(saml2RequestId)) {
            samlRequestIDs.add(saml2RequestId);
            httpSession.setAttribute(SSOAgentConstants.SAML2SSO.ID_ATTRIB_LIST_NAME, samlRequestIDs);
        }

```

Comment (DinikaSen) [reply io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java]: Addressed with https://github.com/asgardeo/asgardeo-java-saml-sdk/pull/22/commits/b673ea42f795da4b22ebf367e76744f40671cf25

File: io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java (hunk 1 above)
Comment (DinikaSen) [reply io.asgardeo.java.saml.sdk/src/main/java/io/asgardeo/java/saml/sdk/SAML2SSOManager.java]: Addressed with https://github.com/asgardeo/asgardeo-java-saml-sdk/pull/22/commits/b673ea42f795da4b22ebf367e76744f40671cf25

**Review by ChamathNS — APPROVED** [review]

**Pull request merged**